### DISCUSSION TITLE
Error SMTP Connection to Mailcow

### DISCUSSION THREAD

**Discussion by JuIez** [dani-garcia/vaultwarden#2721] [Q&A]
Hey Guys, 
I set up Mailcow and Vaultwarden in the same network but for any reasons I can't get an SMTP connection.
I don't even gent anything on my mailcow, it just stucks in vaultwarden.

Even with the help of the other discussions I couln't get it working, any idea what to do

Both are as server:latest and both got build within the last 24 hours.

Thanks in advance :)

![image](https://user-images.githubusercontent.com/44580956/188168208-cd0dff51-c284-4527-984e-620f8c0c03d1.png)
![image](https://user-images.githubusercontent.com/44580956/188168542-1bc6fa9f-5ec1-4dd5-aa30-c2d097a3d3f2.png)

**Comment by BlackDex** [comment]
Looks like Vaultwarden isn't able to access the Mailcow system.
Either Mailcow is running in a private network namespace. So, no exposed ports, and thus Vaultwarden can't connect.
Or, the hostname for your Mailcow can't be resolved in some way.

Check the network config of your container setup, check if the correct ports are exposed where needed.

**Reply by JuIez** [reply]
I put Vaultwarden into the same network as Mailcow and as the system can speak via ports within each other I didn't thought I need to change something there.
![image](https://user-images.githubusercontent.com/44580956/188170839-d66ee01f-d254-488b-a18a-f5edc1c9e675.png)

Otherwise my port 587 is open.

**Reply by BlackDex** [reply]
I can't really help here, not sure what you have configured, but i would suggest to double check the Vaultwarden config then, and provide an IP instead of a hostname maybe.

Other then that it, if Vaultwarden can connect to the system, it will work just fine.

**Reply by JuIez** [reply]
Allright, I'll check that out. 
Thank you :)

**Reply by JuIez** [reply]
Okay, I managed to get an diffrent request error (OS Error 99) but still looking into it. 
Dunno what else to do.

**Reply by BlackDex** [reply]
That looks like an error message regarding duplicate ports or unauthorized to use that port as a non root user maybe.
